### PR TITLE
bugfix/deseng96 Improve logging for better development experience

### PR DESCRIPTION
**Pull request by Baelx**
- **Partial PR**. This doesn't fix the bug but adds developer tools to be able to troubleshoot more effectively.
- Added the ability to silence default logs to be able to focus on one set of operations
- No substantive changes in the following files:
 	modified:   api/controllers/comment.js
	modified:   api/controllers/commentperiod.js
	modified:   api/controllers/document.js
	modified:   api/controllers/emailSubscribe.js
	modified:   api/controllers/list.js
	modified:   api/controllers/login.js
	modified:   api/controllers/organization.js
	modified:   api/controllers/project.js
	modified:   api/controllers/recentActivity.js
	modified:   api/controllers/search.js
	modified:   api/controllers/survey.js
	modified:   api/controllers/surveyResponse.js
	modified:   api/controllers/topic.js
	modified:   api/controllers/user.js
	modified:   api/controllers/vc.js
	modified:   api/helpers/auth.js
	modified:   api/helpers/databaseIndexes.js

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,11 +124,18 @@ mongorestore -d epic dump/[old_database_name_most_likely_esm]
 
 Then run the contents of [dataload](prod-load-db/esm_prod_april_1/dataload.sh) against that database.  You may need to edit the commands slightly to match your db name or to remove the ".gz --gzip" portion if your dump unpacks as straight ".bson" files.
 
-## Developing
+# Developing
```
Comment (ShawnTurple) [inline README.md]: are we using h1 for all headings in readme... when did we make the decision to do this?

File: README.md [hunk 2]
```diff
@@ -237,4 +240,18 @@ oc get routes | grep 'minio'
...
 You will not be able to see the above value of the secret if you try examine it.  You will only see the encrypted values.  Approach your team member with admin access in the openshift project in order to get the access key and secret key values for the secret name you got from the above command.  Make sure to ask for the correct environment (dev, test, prod) for the appropriate values.
+
+## Logging
+
+The `winston` package is used to log nearly every operation in the app. `console.log  ` is discouraged in favour of `winston`.
+
+When developing, one may choose to silence the verbose log output to be able to focus on specific operations. To do this: 
+
+1. Set the SILENCE_DEFAULT_LOG environment variable to "true".
+2. Switch the `winston.loggers.get()` call argument from `defaultLog` to `devLog`.
```
Comment (ShawnTurple) [inline README.md:254]: so are you updating this in code, as oppose to a env variable?, in if it is in code, maybe it should be an env file, if we choose not to use an env, then at least indicate in the readme where this is located..  But env would be preferred.

File: api/controllers/emailSubscribe.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 var auth = require('../helpers/auth');
 var _ = require('lodash');
-var defaultLog = require('winston').loggers.get('default');
+var defaultLog = require('winston').loggers.get('defaultLog');
```
Comment (ShawnTurple) [inline api/controllers/emailSubscribe.js:3]: i'm confused why are we calling this defaultLog ?? why not just log

File: app.js [hunk 4]
```diff
@@ -1,38 +1,30 @@
...
-    console: {
-        colorize: 'true',
-        handleExceptions: true,
-        json: false,
-        level: 'silly',
-        label: 'default',
-    }
-});
-var defaultLog = winston.loggers.get('default');
+const dbUsername = process.env.MONGODB_USERNAME || '';
```
Comment (ShawnTurple) [inline app.js:25]: I worry this is out of scope from the ticket, i understand the reason of the change, i don't understand the why/now of the change.  in actual fact now what i'm seeing is this was the scope, the logging wasn't.

File: config/loggers.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ */
+module.exports.configureAppLogging = () => {
+    /**
+     * Adds a default logger.
+     */
+    winston.loggers.add('defaultLog', {
+        silent: 'true' === process.env.SILENCE_DEFAULT_LOG || false,
+        transports: [
+            new transports.Console({
+            level: 'info',
```
Comment (ShawnTurple) [inline config/loggers.js:23]: this part here might benefit from having an env variable 
```suggestion
            level: process.env.LOGGING_LEVEL || 'info',
```

**Review by CR-gbc — APPROVED** [review]

File: config/loggers.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            format: combine(
+                colorize(),
+                timestamp(new Date().toLocaleString("en-US", {timeZone: "America/Vancouver"})),
+                logFormat
+            )
+            })
+        ],
+    });
+
+    /**
```
Comment (ShawnTurple) [inline config/loggers.js:33]: How did you come about this conclusion, for multiple loggers?
This seems to be slightly convoluted, though could be convinced if there was logical reasons for it.

File: package.json [hunk 7]
```diff
@@ -52,7 +52,7 @@
     "tree-model": "^1.0.7",
     "underscore": "^1.9.1",
     "wicket": "^1.3.3",
-    "winston": "^2.4.4",
+    "winston": "^3.6.0",
```
Comment (ShawnTurple) [inline package.json:55]: again this doesn't seem to be in scope of this ticket.

**Review by ShawnTurple — CHANGES_REQUESTED** [review]
Although technically this could be approved from a coding standpoint, I do have concerns that i think we should discuss.

File: README.md (hunk 1 above)
Comment (Baelx) [reply README.md]: We didn't make a decision for this - but it's a good point you raise if we're concerned about SEO for github pages.

File: README.md (hunk 2 above)
Comment (Baelx) [reply README.md:254]: The ordered list implies a series of steps so both the env needs to get changed, and you need to make a change in your code.

File: app.js (hunk 4 above)
Comment (Baelx) [reply app.js:25]: The logging was certainly more out of scope but it has made my life easier when debugging this and didn't take that long. I can add the time logged to our future ticket for logging improvements.

As for changing var -> const I see this as part of the philosophy of improving coding standards for the files we work on. When possible, and not too much effort in terms of refactoring, I try to make incremental changes. In this case, changing to const lets the developer know that these variables are immutable and helps guard against accidental reassignment for future changes.

File: config/loggers.js (hunk 6 above)
Comment (Baelx) [reply config/loggers.js:33]: The decision was made out of consideration for:
- time to refactor(if I moved to a single logger)
- technical limitations of the Winston API

With this solution, we can keep the "defaultLog" in place throughout the app and let it keep doing what it needs to do(logging almost all operations). If we were to silence it, I don't think the Winston API allows us to "un-silence" specific log messages. The silent flag needs to be set when the logger is instantiated, to my knowledge. Hence the need for one logger that can be silenced, and a separate logger that developers can use when needed.

**Review by ShawnTurple — APPROVED** [review]
